### PR TITLE
Add `enableTracing` option

### LINKED ISSUE #1267: Add enableTracing option
A new simplified options for configuring to turn tracing & performance on should be added to all SDKs. For each SDK the default sample rate should be agreed upon.  

Docs do not need to be updated yet, nor the onboarding wizard. This can wait until the top level options are added to a number of SDKs and we decide if we want to configure the default onboarding experience, for all new users.

See [develop docs](https://develop.sentry.dev/sdk/performance/#enabletracing) for more information. This is a non-breaking change to be clear. Any of the three options should enable performance, alone or in combination

### Open point:
recommend default it 1.0 or 100%. Can discuss if it should be less.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Add `enableTracing` to option

## :bulb: Motivation and Context

Closes #1267

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry_traces_sampler.dart [hunk 1]
```diff
@@ -6,6 +6,9 @@ import '../sentry.dart';
 
 @internal
 class SentryTracesSampler {
+
+  static final defaultSampleRate = 1.0;
```
Comment (marandaneto) [inline dart/lib/src/sentry_traces_sampler.dart]:
```suggestion
  static const _defaultSampleRate = 1.0;
```

**Comment by denrase** [comment]
@marandaneto Sry, pushed to the wrong branch. Will revert

File: dart/lib/src/sentry_traces_sampler.dart [hunk 2]
```diff
@@ -50,11 +52,16 @@ class SentryTracesSampler {
       return parentSamplingDecision;
     }
 
-    final tracesSampleRate = _options.tracesSampleRate;
-    if (tracesSampleRate != null) {
+    double? optionsRate = _options.tracesSampleRate;
+    double? defaultRate = _options.enableTracing == true
+        ? SentryTracesSampler._defaultSampleRate
```
Comment (marandaneto) [inline dart/lib/src/sentry_traces_sampler.dart]:
```suggestion
        ? _defaultSampleRate
```

you probably dont need the class name?

**Comment by marandaneto** [comment]
We probably need to update the docs, need to check which files have to be changed, consider looking at JS/iOS.

**Review by marandaneto — APPROVED** [review]
Thanks @denrase

**Comment by marandaneto** [comment]
@denrase an unit test is failing still.

**Comment by marandaneto** [comment]
> @denrase an unit test is failing still.

@denrase 

> when no tracesSampleRate is set, do not sample with enableTracing null (failed)

**Review by marandaneto — APPROVED** [review]

**Pull request merged**